### PR TITLE
Fix pulp-cli 0.25 compatibility and bump version

### PR DESCRIPTION
**Pull request by quba42**

### CODE REVIEW

**Comment by quba42** [comment]
I will undraft this once I have studied the Pulp CLI 0.25.0 deprecations and performed some basic tests.

**Comment by mdellweg** [comment]
This line should fail:
https://github.com/pulp/pulp-cli-deb/blob/main/pulp-glue-deb/pulp_glue/deb/context.py#L11
How comes it doesn't?

**Comment by mdellweg** [comment]
You need to bump the pulp-glue requirement too too pick up the new version.

File: pyproject.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ classifiers=[
   "Typing :: Typed",
 ]
 dependencies = [
-  "pulp-cli>=0.18.2,<0.25",
+  "pulp-cli>=0.18.2,<0.26",
```
Comment (quba42) [inline pyproject.toml:26]: What is a sane minimum pulp-cli version to ask for?

Comment (quba42) [reply pyproject.toml:26]: I guess I need at leas 0.20.0 if I want to use `specifier` for `PluginRequirement`.

Comment (mdellweg) [reply pyproject.toml:26]: IDK. You can try bisecting to find the best one, or just pick 0.25.0.

**Review by mdellweg — APPROVED** [review]

**Pull request merged**